nx/source/services/applet.c

```diff
@@ -1211,6 +1209,10 @@ bool appletMainLoop(void) {
     }
 
     switch(msg) {
+        case 0x4:
+            appletCallHook(AppletHookType_OnExitRequest);
+        break;
```

Comment:
Add: <code>return false;</code> here. Apps won't need to use appletHook with this, just use appletUnlockExit after main-loop exit. Apps should *not* exit without calling appletUnlockExit if appletLockExit was used.

Reply:
true, that makes that use-case much simpler, I also updated the gist to reflect change